packages/vue/src/composables/useUtils.ts

```diff
@@ -1,33 +1,34 @@
-import { AuthEventData } from '@aws-amplify/ui';
+import {
+  AuthEventData,
+  ValidationError,
+  CodeDeliveryDetails,
+  AuthenticatorRoute,
+  AuthenticatorServiceFacade,
+  SocialProvider,
+  AmplifyUser,
+} from '@aws-amplify/ui';
 
-export const facade = {
+export const facade: AuthenticatorServiceFacade = {
   error: '',
   hasValidationErrors: false,
   isPending: false,
-  route: '',
-  authStatus: '',
-  user: '',
-  validationErrors: [''],
-  codeDeliveryDetails: {
-    val: '',
-    DeliveryMedium: '',
-  },
-  change: (data?: AuthEventData) => null,
+  route: {} as AuthenticatorRoute,
+  authStatus: 'configuring',
+  socialProviders: ['amazon'] as SocialProvider[],
+  initializeMachine: (data?: AuthEventData) => null,
+  user: {} as AmplifyUser,
+  totpSecretCode: null,
+  unverifiedContactMethods: { email: '', phone_number: '' },
+  validationErrors: {} as ValidationError,
+  codeDeliveryDetails: {} as CodeDeliveryDetails,
```

Comment:
Couple things here:
- confused where the default values for `socialProviders`, `user`, `unverifiedContactMethods`, `validationErrors` and `codeDeliveryDetails` are coming from
- we shouldn't need to cast anything in this object since it is typed with `AuthenticatorServiceFacade`

Reply:
I'll remove the casts, except for `AmplifyUser` since that is a large object, with a lot of required fields.

Reply:
[b048bfb](https://github.com/aws-amplify/amplify-ui/pull/3469/commits/b048bfb115b769e358fb6b1374bbef8cfa03838d)

Reply:
Can you explain where the default values are coming from? For example, why would `unverifiedContactMethods` default to `{ email: '', phone_number: '' }`?

Reply:
To follow up here on the casting, just because something is hard to type doesn't mean it's acceptable not to provide the necessary values. The point of TS is to add type safety to JS, and if we are casting just to appease the complier than we lose that safety

Reply:
Understood. The idea here is that were using dummy values so we can use it later in our testing and the keys are used to help loop over the facade and attach them to the the `useAuthenticatorInternal` reactive object.